### PR TITLE
fix/various improvements to Free plan downgrade [GEN-8955]

### PR DESCRIPTION
**Pull request by ecktoteckto**
The following was fixed/improved for the two-step downgrade to plan Free:

**Step 1**
- (fix) the addons that will be removed on downgrade were not shown due to missing parentheses
- (improvement) the text that the downgrade will take place at the end of the current billing cycle is now only shown for orgs billed via Fly

**Step 2 (Survey/final confirmation)**
- (fix) no hint regarding project restart was shown for projects that are created with Micro compute instead of being upgraded to Micro compute after project creation. Resulting in a mismatch between the projects listed in Step 1 and Step 2
- (improvement) the text that the downgrade will happen immediately after clicking confirm was changed for orgs billed via Fly
- (improvement) the success message after clicking confirm was adjusted regarding immediate/scheduled downgrade and projects being restarted
- (improvement) the text regarding credits that will be given is now only shown if the downgrade happens immediately

### CODE REVIEW

File: apps/studio/components/interfaces/Support/Success.tsx [hunk 1]
```diff
@@ -29,8 +38,15 @@ const Success = ({ sentCategory = '' }: SuccessProps) => {
       <div className="flex items-center flex-col space-y-2">
         <h3 className="text-xl">Support request successfully sent!</h3>
         <p className="text-sm text-foreground-light">
-          We will reach out to you at <span className="text-foreground">{respondToEmail}</span>
+          We will reach out to you at <span className="text-foreground">{respondToEmail}</span>.
         </p>
+        {selectedProject !== 'no-project' && (
```
Comment (kevcodez) [inline apps/studio/components/interfaces/Support/Success.tsx:43]: Might be better to check for `project` being set here instead of selectedProject being some string

File: apps/studio/components/interfaces/Support/Success.tsx [hunk 2]
```diff
@@ -29,8 +38,15 @@ const Success = ({ sentCategory = '' }: SuccessProps) => {
...
+          <p className="text-sm text-foreground-light">
+            Your ticket has been logged for the project{' '}
```
Comment (kevcodez) [inline apps/studio/components/interfaces/Support/Success.tsx:45]: Given the customer just selected this, not sure if we're adding any value by repeating what they have selected?

If we add value, project name should be enough, no need to add a ref id or anything

Comment (kevcodez) [reply apps/studio/components/interfaces/Support/Success.tsx:45]: Is this even something you changed? Seems kinda unrelated and might just wrongfully show up here?

File: apps/studio/components/interfaces/Support/SupportForm.tsx [hunk 3]
```diff
@@ -40,9 +40,10 @@ const INCLUDE_DISCUSSIONS = ['Problem', 'Database_unresponsive']
 
 export interface SupportFormProps {
   setSentCategory: (value: string) => void
+  setSelectedProject: (value: string) => void
 }
 
-const SupportForm = ({ setSentCategory }: SupportFormProps) => {
+const SupportForm = ({ setSentCategory, setSelectedProject }: SupportFormProps) => {
```
Comment (kevcodez) [inline apps/studio/components/interfaces/Support/SupportForm.tsx:46]: Is this even something you changed? Seems kinda unrelated and might just wrongfully show up here?

File: apps/studio/components/ui/FilterPopover.tsx [hunk 4]
```diff
@@ -27,6 +27,9 @@ export const FilterPopover = ({
 
   const formattedOptions = activeOptions.map((option) => {
     const base = options.find((x) => x[valueKey] === option)
+    if (!base || !base[labelKey]) {
```
Comment (kevcodez) [inline apps/studio/components/ui/FilterPopover.tsx:30]: Is this even something you changed? Seems kinda unrelated and might just wrongfully show up here?

File: apps/studio/pages/support/new.tsx [hunk 5]
```diff
@@ -82,9 +86,16 @@ const SupportPage = () => {
             ].join(' ')}
           >
             {sentCategory !== undefined ? (
-              <Success sentCategory={sentCategory} />
```
Comment (kevcodez) [inline apps/studio/pages/support/new.tsx:85]: Is this even something you changed? Seems kinda unrelated and might just wrongfully show up here?

File: apps/studio/components/interfaces/Support/Success.tsx (hunk 2 above)
Comment (ecktoteckto) [reply apps/studio/components/interfaces/Support/Success.tsx:45]: No, that's not my changes.

File: apps/studio/components/interfaces/Support/SupportForm.tsx (hunk 3 above)
Comment (ecktoteckto) [reply apps/studio/components/interfaces/Support/SupportForm.tsx:46]: No, that's not my changes.

File: apps/studio/pages/support/new.tsx (hunk 5 above)
Comment (ecktoteckto) [reply apps/studio/pages/support/new.tsx:85]: No, that's not my changes.

File: apps/studio/components/ui/FilterPopover.tsx (hunk 4 above)
Comment (ecktoteckto) [reply apps/studio/components/ui/FilterPopover.tsx:30]: No, that's not my changes.

**Review by kevcodez — APPROVED** [review]
